lib/ansible/executor/playbook_executor.py

```diff
@@ -186,7 +186,7 @@ def run(self):
                             if C.RETRY_FILES_SAVE_PATH:
                                 basedir = C.shell_expand(C.RETRY_FILES_SAVE_PATH)
                             elif playbook_path:
-                                basedir = os.path.dirname(playbook_path)
+                                basedir = os.path.dirname(os.path.realpath(playbook_path))
```

Comment:
```unfrakpath()``` won't work because it calls ```os.path.realpath()``` on the passed in path which is what we need to avoid.  At least in the cases that are failing here, tilde and vars are already substituted by the shell before it is given to ansible-playbook on the commandline so we don't need os.path.expandvars or os.path.expanduser (but they don't hurt either).